tsl/src/bgw_policy/job.c

```diff
@@ -616,6 +617,12 @@ policy_recompression_execute(int32 job_id, Jsonb *config)
 			 "completed recompressing chunk \"%s.%s\"",
 			 NameStr(chunk->fd.schema_name),
 			 NameStr(chunk->fd.table_name));
+
+		/* if hypercore, already reindexed */
+		if (!ts_is_hypercore_am(chunk->amoid))
+		{
+			ts_reindex_recompressed_chunk_concurrent(chunk->table_id);
```

Comment:
Let's add the text from the PR description here as a comment, so that it's clear why we want to reindex.